### ISSUE TITLE
test automation

### ISSUE THREAD

**Issue by FlorianDold** [docmalloc/gplmt#13]
We should have some way to do automated testing.  I'm not sure yet what's the best way to do that, considering that we don't really want to mock too many parts of paramiko or the planetlab api.

A good start might be some tests involving only local workers, but that misses features such as run_per_host or synchronization barriers.

**Comment by docmalloc** [comment]
Definitely ... tests for local functionality is imho the first step e.g. parsing experiment xml descriptions, configuration etc. ... so feel free to go for it

Testing network functionality is a mess since it is hard to trick the remote part or mockup into all possbile error cases...

**Comment by Egomania** [comment]
I agree, that test cases are needed and should be considered right from the start
